test/unittest/transport/TCPv4Tests.cpp

```diff
@@ -2032,7 +2035,7 @@ TEST_F(TCPv4Tests, client_announced_local_port_uniqueness)
 
     std::this_thread::sleep_for(std::chrono::milliseconds(100));
 
-    ASSERT_EQ(receiveTransportUnderTest.get_channel_resources().size(), 2u);
+    ASSERT_GT(receiveTransportUnderTest.get_channel_resources().size(), 2u);
```

Comment:
We could traverse the whole collection, add the values into a set, and check the size of the set to be equal to 2.